bitnami/redis/templates/scripts-configmap.yaml

```diff
@@ -55,7 +55,13 @@ data:
     REDISPORT=$(get_port "$HOSTNAME" "REDIS")
 
     HEADLESS_SERVICE="{{ template "common.names.fullname" . }}-headless.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}"
-    REDIS_SERVICE="{{ template "common.names.fullname" . }}.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}"
+
+    if [ -n "$REDIS_EXTERNAL_MASTER_HOST" ]; then
+        REDIS_SERVICE="$REDIS_EXTERNAL_MASTER_HOST"
```

Comment:
I believe this is making an assumption that won't always hold true. Below, we have:

```bash
    get_sentinel_master_info() {
        ...
        sentinel_info_command="{{- if and .Values.auth.enabled .Values.auth.sentinel }}REDISCLI_AUTH="\$REDIS_PASSWORD" {{ end }}timeout 5 redis-cli -h $REDIS_SERVICE -p $SENTINEL_SERVICE_PORT sentinel get-master-addr-by-name {{ .Values.sentinel.masterSet }}"
        ...
        eval $sentinel_info_command
    }
```

This is making the assumption that sentinel is running on `$SENTINEL_SERVICE_PORT` of `$REDIS_EXTERNAL_MASTER_HOST`.

In our fleet that colocation is true for almost all redises, but one subset (redis-cache) has a dedicated set of sentinels.

If we wanted to support this properly, I think we'd need a separate config setting for `REDIS_EXTERNAL_SENTINEL_HOST` and `REDIS_EXTERNAL_SENTINEL_PORT`. Possibly even allowing multiple ones for HA as a stretch goal.

Alternatively we can bake that assumption into the migration -- then we'd need to move `redis-cache` to colocated redis+sentinel before migrating. Makes things a bit simpler, but is also less flexible.